rawtodigi_alpaka.cc

```diff
@@ -0,0 +1,527 @@
+                uint8_t error = conversionError(fedId, 3, debug); //use the device function and fill the arrays
+                err->push_back(acc, PixelErrorCompact{rawId, ww, error, fedId});
+                if(debug) printf("BPIX1  Error status: %i\n", error);
+                continue;
+              }
+            }
+          } else {
+            // ***conversion rules for dcol and pxid
+            uint32_t dcol = (ww >> pixelgpudetails::DCOL_shift) & pixelgpudetails::DCOL_mask;
+            uint32_t pxid = (ww >> pixelgpudetails::PXID_shift) & pixelgpudetails::PXID_mask;
+            uint32_t row  = pixelgpudetails::numRowsInRoc - pxid/2;
+            uint32_t col  = dcol*2 + pxid%2;
+            localPix.row = row;
+            localPix.col = col;
+            if (includeErrors and not dcolIsValid(dcol, pxid)) {
+              uint8_t error = conversionError(fedId, 3, debug);
+              err->push_back(acc, PixelErrorCompact{rawId, ww, error, fedId});
+              if(debug) printf("Error status: %i %d %d %d %d\n", error, dcol, pxid, fedId, roc);
+              continue;
+            }
+          }
+
+          pixelgpudetails::Pixel globalPix = frameConversion(barrel, side, layer, rocIdInDetUnit, localPix);
+          xx[gIndex]    = globalPix.row;  // origin shifting by 1 0-159
+          yy[gIndex]    = globalPix.col;  // origin shifting by 1 0-415
+          adc[gIndex]   = getADC(ww);
+          pdigi[gIndex] = pack(globalPix.row, globalPix.col, adc[gIndex]);
+          moduleId[gIndex] = detId.moduleId;
+          rawIdArr[gIndex] = rawId;
+        }
+      } // end of loop (gIndex < end)
+      
+    } // end of Raw to Digi kernel
+
+    };
+    
+    void rawtodigi(const Input *input_d, Output *output_d, 
+                  const uint32_t wordCounter,
+                  bool useQualityInfo, bool includeErrors, bool debug, Queue queue) {
+                  
+      Vec const elementsPerThread(Vec::all(static_cast<Idx>(wordCounter)));
+      Vec const threadsPerBlock(Vec::all(static_cast<Idx>(1)));
+      Vec const blocksPerGrid(Vec::all(static_cast<Idx>(1)));
+      //                       (wordCounter + threadsPerBlock[0] - 1) / threadsPerBlock[0])
+      // ));
+
+      WorkDiv const workDiv(
+      blocksPerGrid,
+      threadsPerBlock,
+      elementsPerThread);
+    
+      rawtodigi_kernel<Idx> kernel;
+
+      auto const taskRawToDigiKernel(alpaka::kernel::createTaskKernel<Acc>(
+                                    workDiv,
+                                    kernel,
+                                    input_d,
+                                    output_d,
+                                    useQualityInfo,
+                                    includeErrors,
+                                    debug
+                                    ));
+
+      alpaka::queue::enqueue(queue, taskRawToDigiKernel);
+
+    }     
+    
+  }; // end namespace alpaka
+}; // end namespace ALPAKA_ARCHITECTURE
```

Comment:
The inner namespace is `ALPAKA_ARCHITECTURE`, and outer `Alpaka`, no?

(also, please leave remove the redundant semicolon)